app/scenes/mfa/mfa.js

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {Component} from 'react';
+import {injectIntl, intlShape} from 'react-intl';
+import {TextInput, Image, KeyboardAvoidingView} from 'react-native';
+
+import Button from 'react-native-button';
+import ErrorText from 'app/components/error_text';
+import FormattedText from 'app/components/formatted_text';
+import {GlobalStyles} from 'app/styles';
+
+import logo from 'assets/images/logo.png';
+
+import RequestStatus from 'service/constants/request_status';
+
+class Mfa extends Component {
+    static propTypes = {
+        actions: React.PropTypes.shape({
+            goBack: React.PropTypes.func.isRequired,
+            login: React.PropTypes.func.isRequired
+        }).isRequired,
+        intl: intlShape.isRequired,
+        loginId: React.PropTypes.string.isRequired,
+        password: React.PropTypes.string.isRequired,
+        loginRequest: React.PropTypes.object.isRequired
+    };
+
+    componentWillMount() {
+        this.setState({
+            token: '',
+            error: null
+        });
+    }
+
+    componentWillReceiveProps(nextProps) {
+        // In case the login is successful the previous scene (login) will take care of the transition
+        if (this.props.loginRequest.status === RequestStatus.STARTED &&
+            nextProps.loginRequest.status === RequestStatus.FAILURE) {
+            this.props.actions.goBack();
+        }
+    }
+
+    handleInput = (token) => {
+        this.setState({
+            token,
+            error: null
+        });
+    };
+
+    submit = () => {
+        if (!this.state.token) {
+            this.setState({
+                error: this.props.intl.formatMessage({
+                    id: 'login_mfa.tokenReq',
+                    defaultMessage: 'Please enter an MFA token'
+                })
+            });
+            return;
+        }
+
+        this.props.actions.login(this.props.loginId, this.props.password, this.state.token);
+    };
+
+    render() {
+        const {formatMessage} = this.props.intl;
+
+        return (
+            <KeyboardAvoidingView
+                behavior='padding'
+                style={[GlobalStyles.container, GlobalStyles.signupContainer]}
+            >
+                <Image
+                    source={logo}
+                />
+                <FormattedText
+                    style={[GlobalStyles.header, GlobalStyles.label]}
+                    id='login_mfa.enterToken'
+                    defaultMessage="To complete the sign in process, please enter a token from your smartphone's authenticator"
+                />
+                <ErrorText error={this.state.error}/>
+                <TextInput
```

Comment:
No, it's just a wrapper around `TextInput` that formats the placeholder. It's used in `PostTextbox` like:
```
<TextInputWithLocalizedPlaceholder
    value={this.props.value}
    placeholder={{id: 'create_post.write', defaultMessage: 'Write a message...'}}
    multiline={true}
    style={...}
    ...
/>
```